File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var sum: usize = 0;
    var i: usize = 1;
    while (i <= number) : (i += 1) {
        sum += i;
    }
    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    var squares: usize = 0;
    var i: usize = 1;
    while (i <= number) : (i += 1) {
        squares += i * i;
    }
    return squares;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(n)
    @numbers = 1..n
  end

  def square_of_sum
    @numbers.sum ** 2
  end

  def sum_of_squares
    @numbers.map { |n| n ** 2 }.sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end
